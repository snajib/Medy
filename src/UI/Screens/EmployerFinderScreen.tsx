import React from 'react';
import { View } from 'react-native';
import { styles } from '../Shared/styles';
import { HorizontalScroll } from '../Shared/horizontal-scroll';
import { IEmployee } from '../../../store/app-action-reducer';

export interface IEmployerFinderScreenProps {
  EmployeeList: IEmployee[];
}

export class EmployerFinderScreen extends React.Component<
  IEmployerFinderScreenProps
> {
  render() {
    return (
      <View style={styles.topPrimaryContainterStyle}>
        <HorizontalScroll
          style={styles.finderDefaultCardStyle}
          schema={this.props.EmployeeList.map(employee => employee.name)}
        />
      </View>
    );
  }
}
