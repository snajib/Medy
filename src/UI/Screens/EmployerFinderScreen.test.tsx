import React from 'react';
import renderer from 'react-test-renderer';
import { EmployerFinderScreen } from './EmployerFinderScreen';
import { View } from 'react-native';

describe('EmployerFinderScreen', () => {
  it('should render views correctly', () => {
    const finderScreen = renderer.create(<EmployerFinderScreen />);

    expect(finderScreen.root.findAllByType(View).length).toBe(13);
  });

  it('should render with snapshot', () => {
    const employerFindScreen = renderer.create(<EmployerFinderScreen />);
    expect(employerFindScreen.toJSON()).toMatchSnapshot();
  });
});
