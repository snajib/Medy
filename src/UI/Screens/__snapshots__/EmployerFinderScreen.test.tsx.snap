// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployerFinderScreen should render with snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5F44BA",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    centerContent={true}
    contentContainerStyle={
      Object {
        "marginLeft": 10,
        "marginRight": 10,
      }
    }
    contentInset={
      Object {
        "bottom": 0,
        "left": 0,
        "right": 50,
        "top": 0,
      }
    }
    decelerationRate={0}
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    snapToAlignment="start"
    snapToInterval={670}
  >
    <View>
      <View
        style={
          Object {
            "width": 30,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "height": 1034,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 100,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 3,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text>
            Profile 1
          </Text>
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "height": 1034,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 100,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 3,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text>
            Profile 2
          </Text>
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "height": 1034,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 100,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 3,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text>
            Profile 3
          </Text>
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "height": 1034,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 100,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 3,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text>
            Profile 4
          </Text>
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "height": 1034,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 100,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 3,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text>
            Profile 5
          </Text>
          <Text />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
